content/cosign-verify-bundles.md

```diff
@@ -0,0 +1,106 @@
++++
+title = "cosign Verification of npm Provenance, GitHub Artifact Attestations, and Homebrew Provenance"
+author = "Zach Steindler (GitHub)"
+date = "2024-08-09"
+tags = ["sigstore", "cosign", "softwaresupplychain", "security"]
+type = "post"
+draft = false
++++
+
+One of the feature of the [cosign v2.4.0 release](https://github.com/sigstore/cosign/releases/tag/v2.4.0) allows you to verify attestations in the bundle format used by npm provenance, GitHub Artifact Attestations, and Homebrew provenance.
+
+We'll show how to perform that verification for each ecosystem, and along the way explain some of the details involved. You'll notice these examples follow the same general pattern of getting an artifact to verify, getting the bundle that contains the signed attestation about that artifact, and then providing a verification policy to cosign via command line flags.
+
+To run these examples you'll need up-to-date `curl`, `jq`, `gh`, `brew`, and (of course) `cosign`.
+
+### npm Provenance
+
+[npm provenance](https://docs.npmjs.com/generating-provenance-statements) connects npm packages back to their source code and build instructions. Over 16,000 unique packages have published with provenance, so to start we'll pick one and get an artifact to verify:
+
+```
+$ curl https://registry.npmjs.org/semver/-/semver-7.6.3.tgz > semver-7.6.3.tgz
+```
+
+Then we need the bundle associated with that artifact. npm returns the publish attestations along with the provenance attestation, so we have to pull out just the SLSA provenance (I promise you, I didn't get this right on the first try):
+
+```
+$ curl https://registry.npmjs.org/-/npm/v1/attestations/semver@7.6.3 | jq '.attestations[]|select(.predicateType=="https://slsa.dev/provenance/v1").bundle' > npm-provenance.sigstore.json
+```
+
+Now we're ready to verify! cosign assumes we're using the Sigstore public good instance, which is what npm provenance uses as well. We'll tell cosign what CI/CD system we're expecting built this artifact, via `--certificate-oidc-issuer`, as well as which repository we expect the artifact to come from (and in this case, also which workflow), via `--certificate-identity-regexp`:
+
+```
+$ cosign verify-blob-attestation --bundle npm-provenance.sigstore.json --new-bundle-format --certificate-oidc-issuer="https://token.actions.githubusercontent.com" --certificate-identity-regexp="^https://github.com/npm/node-semver/.github/workflows/release-integration.yml.?" semver-7.6.3.tgz
```

Comment:
Can we call out `--new-bundle-format` explicitly as the addition that makes it possible to support this new bundle format?